li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/largemessage/MessageSplitterImpl.java

```diff
@@ -113,7 +113,8 @@ public MessageSplitterImpl(int maxSegmentSize,
       Headers temporaryHeaders = new RecordHeaders(headers);
       temporaryHeaders.remove(Constants.LARGE_MESSAGE_HEADER);
       LargeMessageHeaderValue largeMessageHeaderValue = new LargeMessageHeaderValue(
-          LargeMessageHeaderValue.LEGACY_V2,
+          // use new version mark
```

Comment:
You should not use the new version mark until every single known client picks up the compatible assembler version. Please continue using `LEGACY_V2`.

Reply:
resolved